### ISSUE TITLE
Fillet can not round a face (upstream OCC bug)

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#5561]
Issue imported from https://tracker.freecad.org/view.php?id=938

* **Reporter:** ghia
* **Date submitted:** 1/1/2013
* **FreeCAD version:** 0.12
* **Category:** Bug
* **Status:** assigned
* **Tags:** #ready-for-migration, fillet, OCC Bug, OCC NotOnRoadmap, upstream

# Original report text

See file in addendum of mantis bug #[937](https://tracker.freecad.org/view.php?id=937)

While the shape is 20 mm thick, you can not place 2 fillets of 10 mm 
Fillet001 which is 9 mm can not be set to 10

# Additional information

This ticket is being tracked from [OCC Bugs in the Bugtracker](https://forum.freecadweb.org/viewtopic.php?f=10&t=20264&p=156236#p156236) forum thread.

# Other bug information

* **Priority:** normal
* **Severity:** minor
* **Category:** Bug
* **Updated:** 1/24/2022

# Discussion from Mantis ticket

### Comment by wmayer 2013-07-02 03:37
Could be a float vs. double precision issue. Should be checked once the double-precision branch is merged into master.

---

### Comment by tanderson69 2013-09-23 17:29
I tried this with the current assembly branch (has float to double change) and there was no change.

I also tried this with the OCC DRAWEXE and witnessed the same behavior.

also searched OCC forum. Here is one thread.
http://www.opencascade.org/org/forum/thread_21317/?forum=3

//OCC rant removed

---

### Comment by wmayer 2015-01-14 13:34
http://tracker.dev.opencascade.org/view.php?id=25478

---

### Comment by Kunda1 2020-12-31 12:02
Can someone test with OCC 7.5.0 ?

Edit: Please remember to update the relationship with Feature #5569 if this ticket is resolved.

**Comment by luzpaz** [comment]
Upstream tickets needs to be bumped and ideally added to OCC roadmap

**Comment by maxwxyz** [comment]
This also applies to the chamfer tool.

**Comment by maxwxyz** [comment]
Is anybody working on this issue? @wwmayer it was automatically assigned to you during the bug migration, shall I unassign you from this?

**Comment by adrianinsaval** [comment]
it might be worth contacting opencascade about pricing to get this feature implemented, then we do a fundraiser to pay for it

**Comment by stefano2734** [comment]
Occ 7.9.2 Update in oct 18 or in 7.9.1 before,  perhaps here a solution to update to last version of occ.

**Comment by VanDoge** [comment]
can we update to occ 7.9.3 the issue should be fixed then
also a bunch of other occ related issues should be fixed then.
champfers make a huge portion of issus

**Comment by maxwxyz** [comment]
@pieterhijma is this true?

**Comment by pieterhijma** [comment]
I tried:

```
OS: Arch Linux (tty)
Architecture: x86_64
Version: 1.2.0dev.44832 (Git)
Build date: 2025/12/30 08:02:30
Build type: Unknown
Branch: main
Hash: 0ced25c6dcf6ec033fb4fc7293bc3bebceb3b31a
Python 3.13.11, Qt 6.10.1, Coin 4.0.5, Vtk 9.5.2, boost 1_89, Eigen3 5.0.1, PySide 6.10.1
shiboken 6.10.1, SMESH 7.7.1.0, xerces-c 3.3.0, OCC 7.9.3
Locale: English/United States (en_US)
Navigation Style/Orbit Style/Rotation Mode: CAD/Trackball/Drag at cursor
Stylesheet/Theme/QtStyle: FreeCAD.qss/FreeCAD Light/
Logical DPI/Physical DPI/Pixel Ratio: 96/92.6073/1.33333
Installed mods: 
  * OpenTheme 2024.5.16
  * fasteners 0.5.34
  * WebTools 1.0.0
  * osh-autodoc-workbench 0.2.3
  * DynamicData 2.70.0
  * Curves 0.6.61
  * freecad.gears 1.3.0
  * sheetmetal 0.7.24
  * BCFPlugin 1.0.0
  * Ondsel-Lens 2025.11.21.01
```

I pulled the original file from [this old mantis bug report](https://tracker.freecad.org/view.php?id=937): [banjo1.FCStd.zip](https://github.com/user-attachments/files/24381513/banjo1.FCStd.zip)

Since that file is broken in current FreeCAD (it has no body and this may be a separate issue), I recreated the file from the sketches: [banjo-recreated.FCStd.zip](https://github.com/user-attachments/files/24381543/banjo-recreated.FCStd.zip)

I tried both fillet and chamfer and it is still the case that they don't work if both values are 10 mm. So, as far as I can see, this has not been fixed in 7.9.3.

**Comment by SayakMukhopadhyay** [comment]
Seems like a fix was merged to the IR branch https://github.com/Open-Cascade-SAS/OCCT/pull/1293

**Comment by stefano2734** [comment]
> Seems like a fix was merged to the IR branch [Open-Cascade-SAS/OCCT#1293](https://github.com/Open-Cascade-SAS/OCCT/pull/1293)

That is in milestone OCCT  8.1 solved issue.  
This version will come after 8.0.2. so perhaps to late to next release FreeCAD 26.3.
https://github.com/Open-Cascade-SAS/OCCT/milestone/6?closed=1

**Comment by maxwxyz** [comment]
No need to comment, already marked on occ board
